datasets_demo/run.jl

```diff
@@ -19,49 +17,157 @@ using DataSets
 # To work with the data in the REPL, we currently have a slightly dubious but
 # really convenient unscoped version of open():
 #
-#   truefx_data = open(BlobTree, dataset("TrueFX"))
-
-using DataFrames
-using CSV
-using CodecZlib
-using Dates
-using Underscores
-
-function load_truefx_csv(csv_blob)
-    @info "Loading $(basename(csv_blob))"
-    open(IO, csv_blob) do io
-        buf = read(GzipDecompressorStream(io))
-        CSV.read(buf, DataFrame,
-                 header=["pair", "timestamp", "bid", "ask"],
-                 dateformat=dateformat"yyyymmdd H:M:S.s",
-                 types=Dict(:date=>DateTime))
+#   truefx_data = open(BlobTree, dataset("HistoricalTrueFX"))
+using Distributed
+@everywhere using DataSets, DataFrames, CSV, CodecZlib, Dates, Underscores, JuMP, GLPK, GLPKMathProgInterface
+@everywhere @eval Base.Filesystem function mktemp(parent::AbstractString=tempdir(); cleanup::Bool=true)
+    isdir(parent) || mkdir(parent)
+    b = joinpath(parent, temp_prefix * "XXXXXX")
+    p = ccall(:mkstemp, Int32, (Cstring,), b) # modifies b
+    systemerror(:mktemp, p == -1)
+    cleanup && temp_cleanup_later(b)
+    return (b, fdio(p, true))
+end
```

Comment:
Haha, this is such a terrible workaround.

Thanks for putting up with it while I figure out a proper fix :sweat_smile:

Reply:
(Proper fix coming, involving https://github.com/c42f/ResourceContexts.jl as a backend)